src/main/scala/calc/Compiler.scala

```diff
@@ -74,6 +78,23 @@ object Compiler {
       case Literal(value) =>
         irt.DoubleLiteral(value)
 
+      case BinaryOp(op, rhs, lhs) =>
+        irt.BinaryOp(binaryOpMap(op), compileExpr(rhs), compileExpr(lhs))
+
+      case Let(name, value, body) =>
+        val irtId = irt.Ident(name.name)
+        irt.Block(List(
+          irt.VarDef(irtId, irtpe.DoubleType, false, compileExpr(value)),
+          compileExpr(body))
+        )
+
+      case Ident(name) =>
+        irt.VarRef(irt.Ident(name))(irtpe.DoubleType)
```

Comment:
I've published a solution, but I don't really know if it's the best way to solve the problem in a pure functional way.